pkg/v3/types/basetypes.go

```diff
@@ -0,0 +1,221 @@
+package types
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+)
+
+// Steps
+// 1. Split packages for types for v2 and v3 -> ocr2keepers PR
+//		In core node, update references. evm20 should point to v2 types and evm21 to v3 types
+// 				-> CL PR
+// 2. Update v3 types iteratively. Should only affect evm21 and not 20. For every PR, open corresponding Cl PR
+//   and verify 21 integration test passes
+
+// Create a new base types package for v3
+
+type UpkeepType uint8
+
+// Add exploratory ticket to add default type (0 value) for unknown
+const (
+	ConditionTrigger UpkeepType = iota
+	LogTrigger
+)
+
+type TransmitEventType int
+
+const (
+	PerformEvent TransmitEventType = iota
+	StaleReportEvent
+	ReorgReportEvent
+	InsufficientFundsReportEvent
+)
+
+type UpkeepState uint8
+
+const (
+	Performed UpkeepState = iota
+	Ineligible
+)
+
+// upkeepID is uint256 on contract
+type UpkeepIdentifier [32]byte
+
+func (u UpkeepIdentifier) String() string {
+	return string(u[:])
```

Comment:
hex encode?

Reply:
going with decimal 10 string as that is used elsewhere in front end, o11y